dlib/serialize.h

```diff
@@ -1655,34 +1707,31 @@ namespace dlib
                 if (objects_read == 0)
                 {
                     throw serialization_error("An error occurred while trying to read the first" 
-                        " object from the file " + filename + ".\nERROR: " + e.info + "\n" + suffix);
+                        " object from the stream.\nERROR: " + e.info + "\n" + suffix);
```

Comment:
Definitely don't remove the filename.  That's a hugely important part of the error message.